### ISSUE TITLE
CA1827 is not reported in an if statement

### ISSUE THREAD

**Issue by marco-bertschi** [dotnet/roslyn-analyzers#7148]
### Analyzer

**Diagnostic ID**: [CA1827](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1827): `Do not use Count()/LongCount() when Any() can be used`

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 8.0.0 (Latest)

### Describe the bug

Telling from the [documentation](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/), I would expect the analyzer to be part of the aforementioned NuGet package.

### Steps To Reproduce

- Create a new Console application with VS 2022, targeting .Net 6.
- add the following code to `Program.cs`
  ```
  // See https://aka.ms/new-console-template for more information
  Console.WriteLine("Hello, World!");
  
  var list = new List<string>();
  
  if(list.Count() == 0)
  {
      Console.WriteLine("empty");
  }
  ```
- Add a new `.editorconfig` file to the solution
  ```
  
  [*.cs]
  #### Benennungsstile ####

  # Benennungsregeln

  dotnet_naming_rule.interface_should_be_begins_with_i.severity = suggestion
  dotnet_naming_rule.interface_should_be_begins_with_i.symbols = interface
  dotnet_naming_rule.interface_should_be_begins_with_i.style = begins_with_i

  dotnet_naming_rule.types_should_be_pascal_case.severity = suggestion
  dotnet_naming_rule.types_should_be_pascal_case.symbols = types
  dotnet_naming_rule.types_should_be_pascal_case.style = pascal_case

  dotnet_naming_rule.non_field_members_should_be_pascal_case.severity = suggestion
  dotnet_naming_rule.non_field_members_should_be_pascal_case.symbols = non_field_members
  dotnet_naming_rule.non_field_members_should_be_pascal_case.style = pascal_case
  
  # Symbolspezifikationen

  dotnet_naming_symbols.interface.applicable_kinds = interface
  dotnet_naming_symbols.interface.applicable_accessibilities = public, internal, private, protected, protected_internal, private_protected
  dotnet_naming_symbols.interface.required_modifiers = 

  dotnet_naming_symbols.types.applicable_kinds = class, struct, interface, enum
  dotnet_naming_symbols.types.applicable_accessibilities = public, internal, private, protected, protected_internal, private_protected
  dotnet_naming_symbols.types.required_modifiers = 

  dotnet_naming_symbols.non_field_members.applicable_kinds = property, event, method
  dotnet_naming_symbols.non_field_members.applicable_accessibilities = public, internal, private, protected, protected_internal, private_protected
  dotnet_naming_symbols.non_field_members.required_modifiers = 

  # Benennungsstile

  dotnet_naming_style.begins_with_i.required_prefix = I
  dotnet_naming_style.begins_with_i.required_suffix = 
  dotnet_naming_style.begins_with_i.word_separator = 
  dotnet_naming_style.begins_with_i.capitalization = pascal_case

  dotnet_naming_style.pascal_case.required_prefix = 
  dotnet_naming_style.pascal_case.required_suffix = 
  dotnet_naming_style.pascal_case.word_separator = 
  dotnet_naming_style.pascal_case.capitalization = pascal_case

  dotnet_naming_style.pascal_case.required_prefix = 
  dotnet_naming_style.pascal_case.required_suffix = 
  dotnet_naming_style.pascal_case.word_separator = 
  dotnet_naming_style.pascal_case.capitalization = pascal_case
  csharp_style_expression_bodied_methods = false:silent
  csharp_style_expression_bodied_constructors = false:silent
  csharp_style_expression_bodied_operators = false:silent
  csharp_style_expression_bodied_properties = true:silent
  csharp_style_expression_bodied_indexers = true:silent
  csharp_style_expression_bodied_accessors = true:silent
  csharp_style_expression_bodied_lambdas = true:silent
  csharp_style_expression_bodied_local_functions = false:silent

  [*.{cs,vb}]
  dotnet_style_coalesce_expression = true:suggestion
  dotnet_style_null_propagation = true:suggestion
  dotnet_style_prefer_is_null_check_over_reference_equality_method = true:suggestion
  dotnet_style_prefer_auto_properties = true:silent
  dotnet_style_object_initializer = true:suggestion
  dotnet_style_collection_initializer = true:suggestion
  dotnet_style_prefer_simplified_boolean_expressions = true:suggestion
  dotnet_style_prefer_conditional_expression_over_assignment = true:silent
  dotnet_style_prefer_conditional_expression_over_return = true:silent
  dotnet_diagnostic.CA1827.severity = error
  ```
  `dotnet_diagnostic.CA1827.severity = error` on the last line is important.

- Add the NuGet packages `Microsoft.CodeAnalysis 4.8.0` and `Microsoft.CodeAnalysis.NetAnalyzers 8.0.0.` via NuGet
- Perform a rebuild

### Expected behavior

In the error list, an error CA1827 is reported on the use of `Count() == 0` instead of `Any()` within the `if` statement of the example program. 

### Actual behavior

No error is reported.

## Additional context

I am not sure if I have added the correct NuGet packages, but as the [description text of Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers) points to a [documentation where CA1827 is mentioned](https://learn.microsoft.com/de-de/dotnet/fundamentals/code-analysis/quality-rules/) I have no idea what I am doing wrong, or what could be done different.

**Comment by mavasani** [comment]
This is by design. [CA1829](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1829) is reported here, not `CA1827`:

![image](https://github.com/dotnet/roslyn-analyzers/assets/10605811/c1ffe643-63d2-417e-85d8-8574cd1f2c7c)
